### PR TITLE
Add unit test and fuzzing against Vyper

### PR DESCRIPTION
**Pull request by MrChico**
runs dapp tests on ci

### CODE REVIEW

File: circle.yml [hunk 1]
```diff
@@ -14,3 +14,14 @@ jobs:
       - run:
           name: Compile the contract
           command: make
+  test:
+    docker:
+      - image: nixorg/nix:circleci
```
Comment (axic) [inline circle.yml:35]: Is this automatically downloading `dapp`?

Comment (MrChico) [reply circle.yml:35]: no, this is getting a docker image with `nix` installed. `dapp` is installed when entering the `nix-shell`

**Comment by axic** [comment]
Please rename the `test` directory to `tests`, because otherwise `make test` gets confused (at least for me). It complains "test" already done.

**Comment by MrChico** [comment]
> Please rename the `test` directory to `tests`, because otherwise `make test` gets confused (at least for me). It complains "test" already done.

Ah, yes. Done.

File: tests/vyper_setup.sol [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
...0191505080610b0052610b009050602060c0825160208401600060025af1610e9d57600080fd5b60c0519050602082610c800101526020810190506000610300600880602084610c0001018260208501600060045af15050805182019150506000601860208206610b800160208284011115610ef157600080fd5b602080610ba082610700600060045af1505081815280905090509050601880602084610c0001018260208501600060045af150508051820191505061084051602082610c0001015260208101905080610c0052610c009050602060c0825160208401600060025af1610f6257600080fd5b60c0519050602082610c8001015260208101905080610c8052610c809050602060c0825160208401600060025af1610f9957600080fd5b60c0519050610ae052606435610ae05114610fb357600080fd5b6001805460018254011015610fc757600080fd5b6001815401815550600154610d0052610d2060006020818352015b60016001610d005116141561101757610ae051610d20516020811061100657600080fd5b600060c052602060c02001556110ab565b6000610d20516020811061102a57600080fd5b600060c052602060c0200154602082610d40010152602081019050610ae051602082610d4001015260208101905080610d4052610d409050602060c0825160208401600060025af161107b57600080fd5b60c0519050610ae052610d00600261109257600080fd5b60028151048152505b8151600101808352811415610fe2575b5050005b5b60006000fd5b61017f6112350361017f60003961017f611235036000f3";
```
Comment (axic) [inline tests/vyper_setup.sol:5]: Can you put in a link with commithash to the eth2.0-specs url here?

File: README.md [hunk 3]
```diff
@@ -15,3 +15,11 @@ The motivation is to run the SMTChecker and the new Yul IR generator option (`--
 5. Finally in the `eth2.0-specs` directory run `make test_deposit_contract` to execute the tests
 
 The Makefile currently compiles the code without optimisations. To enable optimisations add `--optimize` to the `solc` line.
+
+
+## Running randomzied `dapp` tests:
```
Comment (axic) [inline README.md]: -> "randomized"

File: Makefile [hunk 4]
```diff
@@ -2,11 +2,23 @@ all: compile
...
 	@echo -n '{"abi": ' > deposit_contract.json
 	@cat DepositContract.abi >> deposit_contract.json
 	@echo -n ', "bytecode": "0x' >> deposit_contract.json
 	@cat DepositContract.bin >> deposit_contract.json
 	@echo -n '"}' >> deposit_contract.json
+
+
+export DAPP_SKIP_BUILD:=1
+export DAPP_SRC:=.
+export DAPP_JSON:=combined.json
```
Comment (axic) [inline Makefile:22]: Nitpick: Should these be moved under `test:`?

File: tests/vyper_setup.sol [hunk 5]
```diff
@@ -0,0 +1,15 @@
...
+      function write(bytes memory _code) public returns (address target) {
+        assembly {
+            target := create(0, add(_code, 0x20), mload(_code))
+        }
+    }
```
Comment (axic) [inline tests/vyper_setup.sol:9]: Nitpick: can you make indentation consistent across the files? Can run solidity lint. Not important though.

File: circle.yml [hunk 6]
```diff
@@ -1,4 +1,9 @@
+# Note that the following stanza uses CircleCI 2.1 to make use of a Reusable Executor
+# This allows defining a docker image to reuse across jobs.
+# visit https://circleci.com/docs/2.0/reusing-config/#authoring-reusable-executors to learn more.
```
Comment (axic) [inline circle.yml]: Can we remove this? Is it needed. Also should change the version below to `2.1` to be correct.

Comment (MrChico) [reply circle.yml]: removed

**Pull request merged**